### PR TITLE
v-select with combobox option uses search value when blurring

### PR DESCRIPTION
**Pull request by rokkie**
## Description
When using v-select with the `combobox` option and items as objects (not just primitives), and you select an option and then blur the field, the `itemValue` becomes the text value of the selected item object.

## Motivation and Context
See: [Issue](https://github.com/vuetifyjs/vuetify/issues/2878)

## How Has This Been Tested?
Added a new test

## Markup:
See: [Codepen](https://codepen.io/rokkieb/pen/XVpOEg)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [ ] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.

## Comments:
PR submitted to `dev` since the patch applies to code that is not in `master`.

### CODE REVIEW

File: src/components/VSelect/VSelect.js [hunk 1]
```diff
@@ -237,7 +237,10 @@ export default {
         }
       }
     },
-    findExistingItem (item) {
+    findExistingItem (val) {
+      return this.items.find(i => this.valueComparator(this.getValue(i), this.getValue(val)))
```
Comment (johnleider) [inline src/components/VSelect/VSelect.js]: Store `this.getValue(val)` in a variable. It should not be re-evaluated for every find iteration.

File: src/components/VSelect/mixins/select-watchers.js [hunk 2]
```diff
@@ -12,6 +12,11 @@ export default {
       this.$refs.menu && this.$refs.menu.updateDimensions()
     },
     inputValue (val) {
+      // Search for an existing item when a value was selected from the menu
```
Comment (johnleider) [inline src/components/VSelect/mixins/select-watchers.js]: split comment into two lines. bonus points for having second line 2 less characters than the first

File: src/components/VSelect/VSelect-combobox.spec.js [hunk 3]
```diff
@@ -77,6 +77,31 @@ test('VSelect - combobox', ({ mount }) => {
...
+    await wrapper.vm.$nextTick()
+
+    wrapper.vm.selectItem(item)
+    await wrapper.vm.$nextTick()
+
+    wrapper.setData({ isActive: false })
+    await wrapper.vm.$nextTick()
+
+    expect(event).toBeCalledWith(item)
+    expect('Unable to locate target [data-app]').toHaveBeenTipped()
```
Comment (johnleider) [inline src/components/VSelect/VSelect-combobox.spec.js]: Merge in the most recent updates from dev. @jacekkarczmarczyk added better error/warning messages and this test will fail.

File: src/components/VSelect/mixins/select-watchers.js (hunk 2 above)
Comment (rokkie) [reply src/components/VSelect/mixins/select-watchers.js]: Fixed in [f94675dc](https://github.com/rokkie/vuetify/commit/f94675dcaa7163596015a1595831e00255094cf7)

File: src/components/VSelect/VSelect.js (hunk 1 above)
Comment (rokkie) [reply src/components/VSelect/VSelect.js]: Fixed in [7819c6f9](https://github.com/rokkie/vuetify/commit/7819c6f960e02deb5d21782b82dd7d027d00214d)

File: src/components/VSelect/VSelect-combobox.spec.js (hunk 3 above)
Comment (rokkie) [reply src/components/VSelect/VSelect-combobox.spec.js]: I just merged in the most recent (at time of writing) `dev` branch ([0d75322e](https://github.com/rokkie/vuetify/commit/0d75322ebedc87d58f503c8d70cbb7d0ed592db0)) but as far as i can tell the test passes (at least on my machine..). Am i doing it wrong or did things change in the meantime?

Comment (jacekkarczmarczyk) [reply src/components/VSelect/VSelect-combobox.spec.js]: My change add the "[Vuetify]" prefix to the error message, `toHaveBeenTipped` function checks if the console messages **includes** the text from `expect` so it still passes. No need imho to change anything here

**Pull request merged**